simd/src/Kokkos_SIMD_Common_Math.hpp

```diff
@@ -32,8 +36,9 @@ class simd_mask;
 template <class M, class T>
 class const_where_expression;
 
+#ifdef KOKKOS_ENABLE_DEPRECATED_CODE_4
 template <typename T, typename Abi>
-[[nodiscard]] KOKKOS_IMPL_HOST_FORCEINLINE_FUNCTION T
+[[nodiscard]] KOKKOS_DEPRECATED KOKKOS_IMPL_HOST_FORCEINLINE_FUNCTION T
```

Comment:
Can you use `KOKKOS_DEPRECATED_WITH_COMMENT` to provide some guidance on the replacement?